Mathlib/RingTheory/Localization/Integral.lean

```diff
@@ -235,6 +235,52 @@ theorem isIntegral_localization' {R S : Type*} [CommRing R] [CommRing S] {f : R
     (Localization (Submonoid.map (f : R →* S) M)) := Localization.isLocalization
   isIntegral_localization
 
+lemma Algebra.IsIntegral.of_isLocalization (R S Rₚ Sₚ : Type*) [CommRing R] [CommRing S]
+    [CommRing Rₚ] [CommRing Sₚ] [Algebra R S] [Algebra R Rₚ] [Algebra R Sₚ] [Algebra S Sₚ]
+    [Algebra Rₚ Sₚ] [IsScalarTower R S Sₚ] [IsScalarTower R Rₚ Sₚ] (M : Submonoid R)
+    [IsLocalization M Rₚ] [IsLocalization (Algebra.algebraMapSubmonoid S M) Sₚ]
+    [Algebra.IsIntegral R S] :
+    Algebra.IsIntegral Rₚ Sₚ := by
+  refine ⟨fun x ↦ ?_⟩
+  delta IsIntegral
+  convert isIntegral_localization (R := R) (S := S) (M := M) (Rₘ := Rₚ) (Sₘ := Sₚ) x
+  apply IsLocalization.ringHom_ext (M := M)
+  simp [← IsScalarTower.algebraMap_eq]
+
+open nonZeroDivisors Algebra in
+instance IsFractionRing.isLocalization_algebraMapSubmonoid
+    {R S L : Type*} [CommRing R] [CommRing S] [IsDomain R] [Field L]
+    [Algebra R S] [Algebra S L]
+    [IsFractionRing S L] [Algebra.IsIntegral R S] [NoZeroSMulDivisors R S] :
+    IsLocalization (algebraMapSubmonoid S R⁰) L := by
```

Comment:
This is apparently less general than my instance [here](https://github.com/leanprover-community/mathlib4/pull/22966/files#diff-b07b06029f1c92c14d68e21d04720f1c32a7a33fd174abd1b5e543eba6d17411R590-R599) (IsIntegral implies IsAlgebraic and NoZeroSMulDivisors implies FaithfulSMul only under suitable Nontrivial conditions, but these edge cases are easy to deal with). If my PR isn't merge first, feel free to copy that instance.

Reply:
#26316 has been merged